hbase_sql/src/test/java/com/splicemachine/derby/impl/sql/execute/operations/ExternalTableIT.java

```diff
@@ -1010,6 +1010,23 @@ public void testReadTextMismatchSchema() throws Exception {
                             "(_c0 CHAR/VARCHAR(x), _c1 CHAR/VARCHAR(x), _c2 CHAR/VARCHAR(x));'.'",
                     e.getMessage() );
         }
+        methodWatcher.execute("drop table external_t2");
+
+        methodWatcher.executeUpdate(String.format("create external table external_t3 (col1 int, col2 varchar(20) )" +
+                "ROW FORMAT DELIMITED FIELDS TERMINATED BY '|' ESCAPED BY '\\\\' LINES TERMINATED BY '\\n'" +
+                " STORED AS TEXTFILE LOCATION '%s'", file));
+        try {
+            methodWatcher.executeQuery("select * from external_t3");
+            Assert.fail("Exception not thrown");
+        } catch (SQLException e) {
+            Assert.assertEquals("Wrong Exception (" + e.getMessage() + ")", EXTERNAL_TABLES_READ_FAILURE, e.getSQLState());
+            Assert.assertEquals( "wrong exception message",
+                    "External Table read failed with exception '2 attribute(s) defined but 3 present " +
+                            "in the external file : '" + file + "'. Suggested Schema is 'CREATE EXTERNAL TABLE T " +
+                            "(_c0 CHAR/VARCHAR(x), _c1 CHAR/VARCHAR(x), _c2 CHAR/VARCHAR(x));'.'",
+                    e.getMessage() );
+        }
+        methodWatcher.execute("drop table external_t3");
```

Comment:
this should be executed regardless if the test fails or not, think about restructuring the test accordingly (e.g. use `finally`).

Reply:
Hi you're correct for other tests, however here we create many different tables, adding them to the preamble would mean creating 50 tables there, which is - in context of external tables - not really great for overview.

Reply:
ok :)